pilot/pkg/model/context.go

```diff
@@ -817,6 +875,12 @@ const (
 	// InterceptionRedirect implies traffic intercepted by IPtables with REDIRECT mode
 	// This is our default mode
 	InterceptionRedirect TrafficInterceptionMode = "REDIRECT"
+
+	// InterceptionAPI is used for API clients, using ApiListener and higher level processing.
+	// It is similar with NONE - it doesn't generate the iptables filter chains - but has different
+	// response to listener.
+	// This mode also activates returning the high-level Istio and K8S configs.
+	InterceptionAPI TrafficInterceptionMode = "API"
```

Comment:
The interception mode here is meant for standard Envoy proxy. This should not be used as a way of identifying gRPC based xds nodes. These nodes should figure out a way to identify themselves as grpc xds listeners rather than conflating with the standard proxy based interception modes.

Reply:
Interception is not specific to Envoy. 

But I can add an explicit metadata to identify the desired generator. Would that satisfy your concerns ? Interception was just a way to start the discussion :-)